vue-app/src/App.vue

```diff
@@ -96,6 +103,28 @@ export default class App extends Vue {
     await this.$store.dispatch(LOAD_RECIPIENT_REGISTRY_INFO)
   }
 
+  beforeDestroy() {
+    for (const interval of Object.keys(this.intervals)) {
+      clearInterval(this.intervals[interval])
+    }
+  }
+
+  @Watch('$web3.user')
+  loginUser = async () => {
+    if (!this.$web3.user) return
+
+    this.$store.commit(SET_CURRENT_USER, this.$web3.user)
```

Comment:
Won't this overwrite all other user data that $web3 doesn't have (such as `isRegistered` & `brightId`)?

Reply:
I suppose that's ok since we subsequently call `await this.$store.dispatch(LOAD_USER_INFO)`?

Reply:
Well, you are right but we shouldn't reach the case where we overwrite something by mistake. Mainly because this function is only going to be called when `$web3.user` changes. That happens in two moments:

1. When `disconnectWallet` is called: the top guard clause is for this case.
2. When `connectWallet` is called: we are going to overwrite the user but we are calling the rest of the data that completes the user state.